internal/cmdrender/executor.go

```diff
@@ -316,8 +316,33 @@ func fnChain(pl *kptfilev1alpha2.Pipeline, pkgPath types.UniquePath) ([]kio.Filt
 	return runners, nil
 }
 
+// detectPathConflicts returns an error if the same index/path is on multiple resources
```

Comment:
nit: move this function below it's caller `trackInputFiles` and `trackOutputFiles`. It helps with the readability. So when we open the file, reader looks at the top level functions and discovers inward functions later.